diff --git a/doomfeatures.h b/doomfeatures.h
--- a/doomfeatures.h
+++ b/doomfeatures.h
@@ -33,7 +33,7 @@
 
 // Enables sound output
 
-//#undef FEATURE_SOUND
+#define FEATURE_SOUND 1
 
 #endif /* #ifndef DOOM_FEATURES_H */
 
diff --git a/doomgeneric.h b/doomgeneric.h
--- a/doomgeneric.h
+++ b/doomgeneric.h
@@ -5,11 +5,11 @@
 #include <stdint.h>
 
 #ifndef DOOMGENERIC_RESX
-#define DOOMGENERIC_RESX 640
+#define DOOMGENERIC_RESX 320
 #endif  // DOOMGENERIC_RESX
 
 #ifndef DOOMGENERIC_RESY
-#define DOOMGENERIC_RESY 400
+#define DOOMGENERIC_RESY 200
 #endif  // DOOMGENERIC_RESY
 
 
@@ -40,6 +40,7 @@
 void DG_SleepMs(uint32_t ms);
 uint32_t DG_GetTicksMs();
 int DG_GetKey(int* pressed, unsigned char* key);
+int DG_GetMouse(int* buttons, int* delta_x, int* delta_y);
 void DG_SetWindowTitle(const char * title);
 
 #ifdef __cplusplus
diff --git a/i_input.c b/i_input.c
--- a/i_input.c
+++ b/i_input.c
@@ -46,6 +46,7 @@
 // Is the shift key currently down?
 
 static int shiftdown = 0;
+extern int usemouse;
 
 // Lookup table for mapping AT keycodes to their doom keycode
 static const char at_to_doom[] =
@@ -275,12 +276,27 @@
     }
 }
 
+static int AccelerateMouse(int value)
+{
+    const int direction = value < 0 ? -1 : 1;
+    int magnitude = abs(value);
+
+    if (magnitude > mouse_threshold)
+    {
+        magnitude = mouse_threshold + ((magnitude - mouse_threshold) * 2);
+    }
+
+    return magnitude * direction;
+}
 
 void I_GetEvent(void)
 {
     event_t event;
     int pressed;
     unsigned char key;
+    int mouse_buttons;
+    int mouse_delta_x;
+    int mouse_delta_y;
 
     
 	while (DG_GetKey(&pressed, &key))
@@ -323,16 +339,15 @@
         }
     }
 
-
-                /*
-            case SDL_MOUSEMOTION:
-                event.type = ev_mouse;
-                event.data1 = mouse_button_state;
-                event.data2 = AccelerateMouse(sdlevent.motion.xrel);
-                event.data3 = -AccelerateMouse(sdlevent.motion.yrel);
-                D_PostEvent(&event);
-                break;
-                */
+    if (usemouse
+     && DG_GetMouse(&mouse_buttons, &mouse_delta_x, &mouse_delta_y))
+    {
+        event.type = ev_mouse;
+        event.data1 = mouse_buttons;
+        event.data2 = AccelerateMouse(mouse_delta_x);
+        event.data3 = -AccelerateMouse(mouse_delta_y);
+        D_PostEvent(&event);
+    }
 }
 
 void I_InitInput(void)
diff --git a/i_sound.c b/i_sound.c
--- a/i_sound.c
+++ b/i_sound.c
@@ -17,10 +17,6 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-
-#if defined(FEATURE_SOUND) && !defined(__DJGPP__)
-#include <SDL_mixer.h>
-#endif
 
 #include "config.h"
 #include "doomfeatures.h"
diff --git a/v_video.c b/v_video.c
--- a/v_video.c
+++ b/v_video.c
@@ -845,88 +845,6 @@
 
 void V_DrawMouseSpeedBox(int speed)
 {
-    extern int usemouse;
-    int bgcolor, bordercolor, red, black, white, yellow;
-    int box_x, box_y;
-    int original_speed;
-    int redline_x;
-    int linelen;
-
-    // Get palette indices for colors for widget. These depend on the
-    // palette of the game being played.
-
-    bgcolor = I_GetPaletteIndex(0x77, 0x77, 0x77);
-    bordercolor = I_GetPaletteIndex(0x55, 0x55, 0x55);
-    red = I_GetPaletteIndex(0xff, 0x00, 0x00);
-    black = I_GetPaletteIndex(0x00, 0x00, 0x00);
-    yellow = I_GetPaletteIndex(0xff, 0xff, 0x00);
-    white = I_GetPaletteIndex(0xff, 0xff, 0xff);
-
-    // If the mouse is turned off or acceleration is turned off, don't
-    // draw the box at all.
-
-    if (!usemouse || fabs(mouse_acceleration - 1) < 0.01)
-    {
-        return;
-    }
-
-    // Calculate box position
-
-    box_x = SCREENWIDTH - MOUSE_SPEED_BOX_WIDTH - 10;
-    box_y = 15;
-
-    V_DrawFilledBox(box_x, box_y,
-                    MOUSE_SPEED_BOX_WIDTH, MOUSE_SPEED_BOX_HEIGHT, bgcolor);
-    V_DrawBox(box_x, box_y,
-              MOUSE_SPEED_BOX_WIDTH, MOUSE_SPEED_BOX_HEIGHT, bordercolor);
-
-    // Calculate the position of the red line.  This is 1/3 of the way
-    // along the box.
-
-    redline_x = MOUSE_SPEED_BOX_WIDTH / 3;
-
-    // Undo acceleration and get back the original mouse speed
-
-    if (speed < mouse_threshold)
-    {
-        original_speed = speed;
-    }
-    else
-    {
-        original_speed = speed - mouse_threshold;
-        original_speed = (int) (original_speed / mouse_acceleration);
-        original_speed += mouse_threshold;
-    }
-
-    // Calculate line length
-
-    linelen = (original_speed * redline_x) / mouse_threshold;
-
-    // Draw horizontal "thermometer" 
-
-    if (linelen > MOUSE_SPEED_BOX_WIDTH - 1)
-    {
-        linelen = MOUSE_SPEED_BOX_WIDTH - 1;
-    }
-
-    V_DrawHorizLine(box_x + 1, box_y + 4, MOUSE_SPEED_BOX_WIDTH - 2, black);
-
-    if (linelen < redline_x)
-    {
-        V_DrawHorizLine(box_x + 1, box_y + MOUSE_SPEED_BOX_HEIGHT / 2,
-                      linelen, white);
-    }
-    else
-    {
-        V_DrawHorizLine(box_x + 1, box_y + MOUSE_SPEED_BOX_HEIGHT / 2,
-                        redline_x, white);
-        V_DrawHorizLine(box_x + redline_x, box_y + MOUSE_SPEED_BOX_HEIGHT / 2,
-                        linelen - redline_x, yellow);
-    }
-
-    // Draw red line
-
-    V_DrawVertLine(box_x + redline_x, box_y + 1,
-                 MOUSE_SPEED_BOX_HEIGHT - 2, red);
-}
-
+    (void)speed;
+}
+
